lib/nostr.js

```diff
@@ -16,24 +20,52 @@ export const DEFAULT_CROSSPOSTING_RELAYS = [
   'wss://nostr.mutinywallet.com/',
   'wss://relay.mutinywallet.com/'
 ]
+export const DEFAULT_NIP46_RELAYS = [
+  'wss://relay.damus.io/',
+  'wss://relay.snort.social/'
+]
 
 export class Relay {
-  constructor (relayUrl) {
-    const ws = new WebSocket(relayUrl)
+  constructor (relayUrl, reconnectOnClose = false) {
+    this.url = relayUrl
+    this.subs = []
+    this.reconnectOnClose = reconnectOnClose
+    this._closed = false
+    this.error = undefined
+    this.reconnect()
+  }
 
-    ws.onmessage = function (msg) {
+  reconnect () {
+    this._closed = false
+    const ws = new WebSocket(this.url)
+    this.ws = ws
+
+    ws.onmessage = (msg) => {
       const [type, notice] = JSON.parse(msg.data)
       if (type === 'NOTICE') {
         console.log('relay notice:', notice)
       }
     }
 
-    ws.onerror = function (err) {
+    ws.onerror = (err) => {
       console.error('websocket error: ' + err)
       this.error = err
     }
 
-    this.ws = ws
+    ws.onclose = () => {
+      if (this.reconnectOnClose && !this._closed) {
+        setTimeout(() => {
+          this.reconnect()
+        }, 5000)
```

Comment:
Mhh, this only attempts to reconnect once. Is this intended?

I also initially thought there might be a listener leak here because `reconnect` adds listeners again but  I guess since you assign to `onclose`, `onopen` etc. that's not the case. You overwrite them.

Reply:
this reconnects everytime the socket is closed,  on reconnection a new socket is always created, if it is closed again a new socket will be created again an so forth

Reply:
Ahh, makes sense